src/panels/KubectlPanel.ts

```diff
@@ -94,4 +98,32 @@ export class KubectlDataProvider implements PanelDataProvider<"kubectl"> {
     private async handleDeleteCustomCommandRequest(name: string) {
         await deleteKubectlCustomCommand(name);
     }
+
+    private transformCommandForJSONPath(command: string): string {
```

Comment:
💡 Refactor suggestions: (Please do test this post this simplification)

- Use of replaceAll: Instead of manually iterating through characters, `replaceAll` simplifies the process of replacing both single and double quotes.
- Arrow Function: Used for `escapeJsonpath` to make the code more concise and modern.
- Removed unnecessary variable: Directly returning the result from `replace`, avoiding the extra `transformedCommand` variable.

```*.ts
private transformCommandForJSONPath(command: string): string {
    // Regular expression to match JSONPATH expressions
    const jsonpathRegex = /-o jsonpath='([^']+)'/g;

    // Function to escape JSONPATH expression
    const escapeJsonpath = (jsonpath: string): string =>
        jsonpath.replaceAll('"', '\\"').replaceAll("'", '"');

    // Replace JSONPATH expressions in the command
    return command.replace(jsonpathRegex, (_match, jsonpath) => {
        const escapedJsonpath = escapeJsonpath(jsonpath);
        return `-o jsonpath="${escapedJsonpath}"`;
    });
}

```

Reply:
Ah good advice! I included this change in latest, and have tested it as well